File: AutomaticBrewingCoffeeBE/AutomaticBrewingCoffee.Main/Domains/Models/ProductCategory.cs
using System.ComponentModel.DataAnnotations;

namespace AutomaticBrewingCoffee.Domain.Models;

public class ProductCategory : BaseModel
{
    [StringLength(50)] [Required] public string ProductCategoryId { get; set; } = null!;

    [Required] [StringLength(100)] public string Name { get; set; } = null!;

    [StringLength(450)] public string? Description { get; set; }

    [StringLength(10)] [Required] public string Status { get; set; } = null!;
    
    public string? ImageUrl { get; set; }
    
    public virtual IEnumerable<Product>? Products { get; set; }
    
    public int? DisplayOrder { get; set; } = 0;
    
}

File: AutomaticBrewingCoffeeApp/lib/app/data/models/product/product_category.dart
class ProductCategory {
  final String productCategoryId;
  final String name;
  final String description;
  final String status;
  final String imageUrl;
  final int displayOrder;

  ProductCategory({
    required this.productCategoryId,
    required this.name,
    required this.description,
    required this.status,
    required this.imageUrl,
    this.displayOrder = 0,
  });

  // Create from JSON
  factory ProductCategory.fromJson(Map<String, dynamic> json) {
    return ProductCategory(
      productCategoryId: json['productCategoryId'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      status: json['status'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      displayOrder: json['displayOrder'] != null
          ? int.tryParse(json['displayOrder'].toString()) ?? 0
          : 0,
    );
  }

  // Convert to JSON
  Map<String, dynamic> toJson() {
    return {
      'productCategoryId': productCategoryId,
      'name': name,
      'description': description,
      'status': status,
      'imageUrl': imageUrl,
      'displayOrder': displayOrder,
    };
  }
}